server/routes/varyingLicences/handlers/policyChange.ts

```diff
@@ -99,62 +99,89 @@ export default class PolicyChangeRoutes {
       additionalLicenceConditions = licence.additionalPssConditions
     }
 
-    const inputs = []
-    let licenceConditionCodes = additionalLicenceConditions.map((c: AdditionalCondition) => c.code)
+    const inputs: string[] = []
+    const licenceConditionCodes = additionalLicenceConditions.map((c: AdditionalCondition) => c.code)
+    const conditionsToAdd: PolicyAdditionalCondition[] = []
 
     // if the condition is being replaced by new condition(s)
     if (condition.changeType === conditionChangeType.DELETED || condition.changeType === conditionChangeType.REPLACED) {
       const replacedByCodes = condition.suggestions.map((replacement: { code: string }) => replacement.code)
 
       // Remove replaced condition and any previously-selected replacements
       // Previously selected replacements removed to allow user to go back and unselect or reselect erroneous changes
-      licenceConditionCodes = licenceConditionCodes
-        .filter((code: string) => code !== condition.code)
-        .filter((code: string) => !replacedByCodes.includes(code))
+      const conditionsToDelete = licenceConditionCodes
+        .filter(code => code === condition.code || replacedByCodes.includes(code))
+        .filter(code => !req.body.additionalConditions?.includes(code))
+
+      this.licenceService.deleteAdditionalConditionsByCode(conditionsToDelete, licence.id, user)
 
       // Add replacement conditions
       if (req.body.additionalConditions?.length > 0) {
-        const replacementArray = await Promise.all(
-          req.body.additionalConditions.map((code: string) => {
-            return this.conditionService.getAdditionalConditionByCode(code, licence.version)
+        await Promise.all(
+          await req.body.additionalConditions.map(async (code: string) => {
+            const conditionToAdd = await this.conditionService.getAdditionalConditionByCode(code, licence.version)
```

Comment:
**question:** Is it ok to use nested async/await?

Reply:
I believe with the use of `map` to return the `Promise` from each iteration, then resolving the returned array of Promises with `Promise.all()`, it works!

Reply:
ok.
Thank you.